### PR TITLE
feat(go): support new features with go code

### PR DESCRIPTION
**Pull request by Reanon**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/go/chapter_backtracking/subset_sum_ii.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Created Time: 2023-06-24
+// Author: Reanon (793584285@qq.com)
+
+package chapter_backtracking
+
+import "sort"
+
+type subsetII struct{}
+
+/* 回溯算法：子集和 I */
```
Comment (krahets) [inline codes/go/chapter_backtracking/subset_sum_ii.go:11]: 子集和 II

File: codes/go/chapter_heap/top_k.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	return last
+}
+
+// Top 获取堆顶元素
+func (h *minHeap) Top() any {
+	return (*h)[0]
+}
+
+func topKHeap(nums []int, k int) *minHeap {
+	// intHeap 为大顶堆
```
Comment (krahets) [inline codes/go/chapter_heap/top_k.go:34]: intHeap 是什么

File: codes/go/chapter_backtracking/subset_sum_ii.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+// File: subset_sum_i.go
```
Comment (krahets) [inline codes/go/chapter_backtracking/subset_sum_ii.go:1]: ii

**Review by krahets — APPROVED** [review]
Thanks @Reanon ! Please address the comment

File: codes/go/chapter_backtracking/subset_sum_ii.go (hunk 1 above)
Comment (Reanon) [reply codes/go/chapter_backtracking/subset_sum_ii.go]: fixed

File: codes/go/chapter_heap/top_k.go (hunk 2 above)
Comment (Reanon) [reply codes/go/chapter_heap/top_k.go]: 笔误，Fixed

File: codes/go/chapter_backtracking/subset_sum_ii.go (hunk 3 above)
Comment (Reanon) [reply codes/go/chapter_backtracking/subset_sum_ii.go]: Fixed

**Comment by krahets** [comment]
`built_in_hash.go` has not been finished yet. Could you help with it? @Reanon

**Comment by Reanon** [comment]
> `built_in_hash.go` has not been finished yet. Could you help with it? @Reanon

get

**Pull request merged**